### ISSUE TITLE
BottomNavigationBar should show stroke icons for unselected items, filled icon for the selected item

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#16574]
In order to make it easier for people with color blindness to determine which tab is selected in the BottomNavigationBar with style "fixed" the selected tab should show a filled icon whereas all unselected tabs should show a stroked icon.

**Comment by goderbauer** [comment]
Not sure if we can automatically determine the filled icon based on a stroked icon or vice versa.

We probably need to extent https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/BottomNavigationBarItem-class.html to take two separate icons for selected and unselected state?

**Comment by jonahwilliams** [comment]
Not even all material icons have stroked/filled variants - was there any specific recommendations here or more guidance given?

**Comment by dnfield** [comment]
I suspect this PR could help here: https://github.com/flutter/engine/pull/5395

I believe if you draw the glyph with paint style as stroke, you'll get the desired effect.

**Comment by dnfield** [comment]
For example:

```
ui.ParagraphBuilder pb = new ui.ParagraphBuilder(new ui.ParagraphStyle());
    pb.pushStyle(new ui.TextStyle(
        fontFamily: icon.fontFamily,
        foreground: new Paint()
          ..color = Colors.black
          ..style = PaintStyle.stroke));
    pb.addText(new String.fromCharCode(icon.codePoint));
    var paragraph = pb.build();
    paragraph.layout(new ui.ParagraphConstraints(width: double.infinity));
    y += 20.0;
    canvas.drawParagraph(paragraph, new Offset(10.0, y));
```

will draw the icon stroked instead of filled, whether there's a stroked glyph or not in the font.

**Comment by dnfield** [comment]
You could set a `PaintStyle` based on whether the icon is selected or not.

**Comment by jonahwilliams** [comment]
From offline discussion, the resolution appears to be that stroke/filled distinction _can_ make things more accessible only if there are stroked/filled variants already and is only required if there are no labels.  Adding support for an alternative icon and some gallery examples should resolve this.

@dnfield I'm not really sure that is something we should be trying to do for the users automatically, especially since the icon property takes any Widget (like an ImageIcon).

**Comment by jonahwilliams** [comment]
Taking a look at some of the icons here - https://material.io/tools/icons/?style=baseline - I think that approach wouldn't work for many of them

**Comment by dnfield** [comment]
Ahhh yeah.  This would only work for font glyphs.
